android/authentication.md

```diff
@@ -1,615 +1,704 @@
 
-* To change the font in the sign-in views, use the `fontFamily` method and pass in the string that represents a font family.
+## Install Dependencies
 
-* To draw the `backgroundColor` full screen, use `fullScreenBackgroundColor`.
+After initialization in your project directory with `amplify init`, update your **App** `build.gradle` with the below:
 
-```java
-import android.app.Activity;
-import android.graphics.Color;
-import android.os.Bundle;
+```groovy
+//For AWSMobileClient only:
+implementation 'com.amazonaws:aws-android-sdk-mobile-client:2.+'
 
-import com.amazonaws.mobile.auth.facebook.FacebookButton;
-import com.amazonaws.mobile.auth.google.GoogleButton;
-import com.amazonaws.mobile.auth.ui.AuthUIConfiguration;
-import com.amazonaws.mobile.auth.ui.SignInUI;
+//For the drop-in UI also:
+implementation 'com.amazonaws:aws-android-sdk-auth-userpools:2.+'
+implementation 'com.amazonaws:aws-android-sdk-auth-ui:2.+'
+```
 
-import com.amazonaws.mobile.client.AWSMobileClient;
-import com.amazonaws.mobile.client.AWSStartupHandler;
-import com.amazonaws.mobile.client.AWSStartupResult;
+For the `AWSMobileClient` alone you can have a minimum SDK version of **15**, but for the drop-in UI you will need a minimum of **23** set in your `build.gradle`:
 
-class MainActivity : AppCompatActivity() {
-    override fun onCreate(savedInstanceState : Bundle?) {
-        super.onCreate()
-        AWSMobileClient.getInstance().initialize(this) {
-            val config = AuthUIConfiguration.Builder()
-                    .userPools(true) // show the Email and Password UI
-                    .signInButton(FacebookButton.class) // Show Facebook
-                    .signInButton(GoogleButton.class) // Show Google
-                    .logoResId(R.drawable.mylogo) // Change the logo
-                    .backgroundColor(Color.BLUE) // Change the background color
-                    .isBackgroundColorFullScreen(true) // Full screen background color
-                    .fontFamily("sans-serif-light") // font
-                    .canCancel(true) // Add a cancel/back button
-                    .build()
-            val signInUI = AWSMobileClient.getInstance().getClient(this@MainActivity, SignInUI::class.java) as SignInUI
-            signInUI.login(this@MainActivity, NextActivity::class.java).authUIConfiguration(config).execute()
-        }.execute()
-    }
-}
 ```
-</div>
-
+minSdkVersion 23
```

Comment:
Prefer for this to be 15. Quick visual inspection may deter people from using mobile client. The IDE or compile will tell them they need 23 when adding the drop-in UI piece.

Reply:
Agree. Fixed.